lm_eval/tasks/ja/mgsm.py

```diff
@@ -0,0 +1,159 @@
+
+        # if ctx fits in max length, return
+        if len(self._tokenize(ctx)) <= max_length:
+            return ctx
+
+        # if ctx is too long, split on a tag that separates each example
+        ctxs = ctx.split(question_tag)
+
+        # if there is no example and still the prompt is too long, fail
+        if len(ctxs) < 2:
+            raise ValueError(f"0-shot description+question doesn't fit in max length. ctx: {ctx}")
+        
+        # delete the first example
+        del ctxs[1]
+
+        # recurse
+        return self.preprocess_ctx(question_tag.join(ctxs), max_length, question_tag)
+
+    def _tokenize(self, text, **kwargs):
+        encode_fn = self.tokenizer.encode
+        if "add_special_tokens" in inspect.getfullargspec(encode_fn).args:
+            encode_params = dict(add_special_tokens=False)
+        else:
+            encode_params = {}
+        return encode_fn(text, **encode_params, **kwargs)
+
+    def _extract_answer(self, completion):
+        matches = ANS_RE.findall(completion)
+        if matches:
+            match_str = matches[-1].strip('.')
+            match_str = match_str.replace(",", "")
+            match_float = float(match_str)
+            if match_float.is_integer():
+                return int(match_float)
+
+        return INVALID_ANS
+
+
+    def process_results(self, doc, results):
+        """Take a single document and the LM results and evaluates, returning a
+        dict where keys are the names of submetrics and values are the values of
+        the metric for that one document
+
+        :param doc:
+            The document as returned from training_docs, validation_docs, or test_docs.
+        :param results:
+            The results of the requests created in construct_requests.
+        """
+        assert len(results) == 1, f"results should be a list with 1 str element, but is {results}"
+        completion = results[0]
+        extracted_answer = self._extract_answer(completion)
+        answer = doc["answer_number"]
+        acc = extracted_answer == answer
+        out = {"acc": acc}
+        out["details"] = {"question": doc["question"], "context": doc["context"], 
+                          "completion": completion, "extracted_answer": extracted_answer,
+                          "answer": answer, "acc": acc}
+        return out
+
+
+class MGSMWithRinnaInstructionSFT(MGSM):
+    """
+    Reference:
+    - HF Hub: https://huggingface.co/rinna/japanese-gpt-neox-3.6b-instruction-sft
+    """
+    PROMPT_VERSION = 0.4
+    SEP = "<NL>"
+    DESCRIPTION = f"ユーザー: 与えられた問題をステップごとに解説してください。{SEP}システム: 分かりました。"
+
+    def doc_to_text(self, doc):
+        input_text = f"問題：{doc['question'].replace('問題：','')}"
+        return f"{self.SEP}ユーザー: {input_text}{self.SEP}システム: ステップごとの答え："
+    
+    def preprocess_ctx(self, ctx, max_length, question_tag="問題："):
+        # the default few_shot constructor adds newlines that are not needed for rinna
+        ctx = ctx.replace('\n', '')
```

Comment:
maybe should we replace to `<NL>` instead of null string? 
```suggestion
        ctx = ctx.replace('\n', self.SEP)
```

Reply:
Obsolete by new fewshot_sep implementation